In [1]:
import pandas as pd
import sys

# set the python path to the directory where the script with funtios are stored
sys.path.insert(0, '/Users/jijovarghese/Projects/Portfolio/Zillow/pyscripts')

#importing function from the ipynb.fs.full.pyfunctions library
from ipynb.fs.full.pyfunctions import read_file

In [16]:
!ls -lh extract/

total 2695104
-rw-r--r--@ 1 jijovarghese  staff   619M 11 Dec 04:28 properties_2016.csv
-rw-r--r--@ 1 jijovarghese  staff   620M 11 Dec 04:29 properties_2017.csv
-rw-r--r--@ 1 jijovarghese  staff    60M 11 Dec 04:30 sample_submission.csv
-rw-r--r--@ 1 jijovarghese  staff   2.3M 11 Dec 04:30 train_2016_v2.csv
-rw-r--r--@ 1 jijovarghese  staff   2.7M 11 Dec 04:30 train_2017.csv
-rw-r--r--@ 1 jijovarghese  staff    19K 11 Dec 04:30 zillow_data_dictionary.xlsx


In [3]:
# Read Properties file
properties_2016 = read_file('properties_2016',verbose=True)

/Applications/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2901: DtypeWarning: Columns (22,32,34,49,55) have mixed types. Specify dtype option on import or set low_memory=False.
  if self.run_code(code, result):



******************** Reading in the properties_2016 dataset ********************

it has 2985217 rows and 58 columns

************************* It has the following columns *************************

Index(['parcelid', 'airconditioningtypeid', 'architecturalstyletypeid',
       'basementsqft', 'bathroomcnt', 'bedroomcnt', 'buildingclasstypeid',
       'buildingqualitytypeid', 'calculatedbathnbr', 'decktypeid',
       'finishedfloor1squarefeet', 'calculatedfinishedsquarefeet',
       'finishedsquarefeet12', 'finishedsquarefeet13', 'finishedsquarefeet15',
       'finishedsquarefeet50', 'finishedsquarefeet6', 'fips', 'fireplacecnt',
       'fullbathcnt', 'garagecarcnt', 'garagetotalsqft', 'hashottuborspa',
       'heatingorsystemtypeid', 'latitude', 'longitude', 'lotsizesquarefeet',
       'poolcnt', 'poolsizesum', 'pooltypeid10', 'pooltypeid2', 'pooltypeid7',
       'propertycountylandusecode', 'propertylandusetypeid',
       'propertyzoningdesc', 'rawcensustractandblock', 'regionidcity

In [4]:
#Read train file
train_2016 = read_file('train_2016_v2',verbose=True)


********************* Reading in the train_2016_v2 dataset *********************

it has 90275 rows and 3 columns

************************* It has the following columns *************************

Index(['parcelid', 'logerror', 'transactiondate'], dtype='object')

*********************** The first 5 rows look like this ************************

   parcelid  logerror transactiondate
0  11016594    0.0276      2016-01-01
1  14366692   -0.1684      2016-01-01
2  12098116   -0.0040      2016-01-01
3  12643413    0.0218      2016-01-02
4  14432541   -0.0050      2016-01-02


In [5]:
#Records missing from train dataset while comparing with properties
len(set(train_2016.parcelid) - set(properties_2016.parcelid))

0

In [6]:
#Records missing from properties dataset while comparing with train
len(set(properties_2016.parcelid) - set(train_2016.parcelid))

2895067

There are 2895067 records in propertis dataset with no match in train dataset. We will merge the dataframes by 
removing the unmatched records

In [8]:
#merge datasets by removing unmatching records from properties
merged_df = train_2016.merge(properties_2016,how='left',on='parcelid')

In [9]:
#Save the merged dataset to csv file for futher analysis
!mkdir -p formatted_data
merged_df.to_csv('formatted_data/train_prop_merged_2016.csv', index=False)